include/mata/utils/partition-relation-pair.hh

```diff
@@ -0,0 +1,1449 @@
+ * Likewise, the node 2 contains indices 2 (first) and 4 (last) which means that
+ * the blockItems 2 and 4 surround a contiguous subvector of elements in the
+ * node 2 (or in the block 1).
+ * Moreover, we also represent the former block which does not exist anymore
+ * by the node 0 which contains indices 0 (first) and 4 (last) which means that
+ * the blockItems 0 and 4 surround a contiguous subvector of elements in the 
+ * node 0. Thus, we know that there had been a block {0, 1, 2, 3, 4} before 
+ * it has been split to obtain blocks {0, 2} and {1, 3, 4}.
+ * 
+ *
+ *             0       1       2       3       4
+ *             ------- ------- ------- ------- -------
+ *            |   0   |   2   |   1   |   4   |   3   |    m_states
+ *             ------- ------- ------- ------- -------
+ *                ↑          ↑ ↑             ↑ ↑
+ *                |          \ /             \ /
+ *                |           X               X
+ *                |          / \             / \
+ *             0  ↓    1     ↓ ↓    2   3    ↓ ↓     4
+ *             ------- ------- ------- ------- -------
+ *            |   0   |   2   |   1   |   4   |   3   |    m_blockItems
+ *       --→→→|-------|-------|-------|-------|--------←←←------------
+ *       |    |   0   |   0   |   1   |   1   |   1   |               |
+ *       |     ------- ------- ------- ------- -------                |
+ *       |        |       |       |       |       |                   |
+ *       |     0  ↓       ↓    1  ↓       ↓       ↓                   |
+ *       |     ----------------------------------------               |
+ *       |    |       1       |            2           |   m_blocks   |
+ *       |     ----------------------------------------               |
+ *       |                |       |                                   |
+ *       |     0       1  ↓    2  ↓                                   |
+ *       |     ------- ------- -------                                |
+ *       -----|   0   |   0   |   2   |   m_nodes                     |
+ *            |-------|-------|-------|                               |
+ *            |   4   |   1   |   4   |                               |
+ *             ------- ------- -------                                |
+ *                |                                                   |
+ *                ----------------------------------------------------
+ * 
+ *  Using this structure, we can:
+ *  - find the block which contains given state in O(1)
+ *  - find a representative state of the given block in O(1)
+ *  - test whether two states share the same block in O(1)
+ *  - test whether all states in a vector A share the same block in O(|A|)
+ *  - iterate through the block B in O(|B|)
+ *  - split the whole partition such that each block
+ *    is split in two pieces or remains unchanged in O(|S|)
+ *  - remember all ancestors of current blocks and access them
+ *
+ */
+typedef struct Partition
```

Comment:
Thanks for your comment. I originally decided not to split the class into two files since I had been asked not to create many files. However, I am able to split it if the others agree.